src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -200,40 +196,69 @@ public static NodeSettings Default(Network network = null, ProtocolVersion proto
         }
 
         /// <summary>
-        /// Loads the configuration file.
+        /// Creates the configuration file if it does not exist.
         /// </summary>
         /// <param name="features">The features to include in the configuration file if a default file has to be created.</param>
-        /// <returns>Initialized node configuration.</returns>
-        /// <exception cref="ConfigurationException">Thrown in case of any problems with the configuration file or command line arguments.</exception>
-        public NodeSettings LoadConfiguration(List<IFeatureRegistration> features = null)
+        /// <returns>Returns <c>true</c> if the file was created and <c>false</c> otherwise.</returns>
+        public bool CreateDefaultConfigurationFile(List<IFeatureRegistration> features)
         {
-            // Configuration already loaded?
-            if (this.ConfigReader != null)
-                return this;
+            // If the config file does not exist yet then create it now.
+            if (this.ConfigurationFile != null && !File.Exists(this.ConfigurationFile))
+            {
+                this.Logger.LogDebug("Creating configuration file '{0}'.", this.ConfigurationFile);
 
-            // Get the arguments set previously
-            var args = this.LoadArgs;
+                File.WriteAllText(this.ConfigurationFile, this.GetDefaultConfiguration(features));
+                this.SetCombinedConfiguration();
+                this.LoadConfiguration();
 
-            // If no configuration file path is passed in the args, load the default file.
-            if (this.ConfigurationFile == null)
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Sets the ConfigReader based on the arguments and the configuration file if it exists.
+        /// Also sets the logging settings for the configuration.
+        /// </summary>
+        private void SetCombinedConfiguration()
+        {
+            // Read the configuration file if it exists.
+            string configText = "";
+            if (this.ConfigurationFile != null && File.Exists(this.ConfigurationFile))
             {
-                this.ConfigurationFile = this.CreateDefaultConfigurationFile(features);
+                this.Logger.LogDebug("Reading configuration file '{0}'.", this.ConfigurationFile);
+                configText = File.ReadAllText(this.ConfigurationFile);
             }
+            else
+                this.Logger.LogDebug("No configuration file. Proceeding with defaults.");
+
+            // Get the arguments set previously.
+            var args = this.LoadArgs;
 
             // Add the file configuration to the command-line configuration.
-            var fileConfig = new TextFileConfiguration(File.ReadAllText(this.ConfigurationFile));
+            var fileConfig = new TextFileConfiguration(configText);
```

Comment:
if we get into the `else` above, configText will be "".
Previously we created the default config file if we didn't have one.

Reply:
In practice the default configuration was only ever created from the FullNodeBuilder where we had the list of features available. Creating an empty configuration file is the only thing we are no longer doing as we would be creating the file from the FullNodeBuilder anyway.